spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/XsuaaJwtDecoder.java

```diff
@@ -132,10 +130,11 @@ private Jwt verifyWithOnlineKey(String token, String jku, String kid) {
 
 	// TODO extract into separate class / bean
 	private JwtDecoder getDecoder(String jku) {
-		NimbusJwtDecoderJwkSupport decoder = new NimbusJwtDecoderJwkSupport(jku);
-		Optional.ofNullable(restOperations).ifPresent(decoder::setRestOperations);
-		decoder.setJwtValidator(tokenValidators);
-		return decoder;
+		NimbusJwtDecoder.JwkSetUriJwtDecoderBuilder jwkSetUriJwtDecoderBuilder = NimbusJwtDecoder.withJwkSetUri(jku);
+		if (restOperations != null) {
+			jwkSetUriJwtDecoderBuilder.restOperations(restOperations);
+		}
```

Comment:
setJwtValidator missing... 

there should be a test that makes sure, that all validators (timestamp, audience, ... ) that are specified with XsuaaJwtDecoderBuilder are called.

Reply:
see comment above